### PR TITLE
boards: arm: gd32f450z_eval: add initial support

### PR DESCRIPTION
**Pull request by sbourdelin**
Add initial support for the GD32F450Z-EVAL board.

Signed-off-by: Sebastien Bourdelin <sebastien.bourdelin@rtone.fr>

### CODE REVIEW

File: boards/arm/gd32f450z_eval/gd32f450z_eval.dts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	status = "okay";
+};
+
+&gpiod {
+	status = "okay";
+};
+
+&gpioe {
+	status = "okay";
+};
```
Comment (gmarull) [inline boards/arm/gd32f450z_eval/gd32f450z_eval.dts:90]: this one doesn't seem to be used

File: boards/arm/gd32f450z_eval/gd32f450z_eval.dts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+&gpiof {
+	status = "okay";
+};
```
Comment (gmarull) [inline boards/arm/gd32f450z_eval/gd32f450z_eval.dts:94]: same here

File: boards/arm/gd32f450z_eval/gd32f450z_eval.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+type: mcu
+arch: arm
+ram: 256
+flash: 3072
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
+supported:
+  - pwm
```
Comment (gmarull) [inline boards/arm/gd32f450z_eval/gd32f450z_eval.yaml:15]: this is missing a few entries (I see other boards should be fixed as well)
- gpio
- dac
- spi
- i2c

this is used by the testing infrastructure

**Review by gmarull — COMMENTED** [review]
lgtm, a few minor nitpicks

File: boards/arm/gd32f450z_eval/gd32f450z_eval.dts (hunk 1 above)
Comment (sbourdelin) [reply boards/arm/gd32f450z_eval/gd32f450z_eval.dts:90]: indeed, thx!

File: boards/arm/gd32f450z_eval/gd32f450z_eval.dts (hunk 2 above)
Comment (sbourdelin) [reply boards/arm/gd32f450z_eval/gd32f450z_eval.dts:94]: ditto

File: boards/arm/gd32f450z_eval/gd32f450z_eval.yaml (hunk 3 above)
Comment (sbourdelin) [reply boards/arm/gd32f450z_eval/gd32f450z_eval.yaml:15]: yes thx, i should probably add `uart` too

**Review by gmarull — DISMISSED** [review]

File: boards/arm/gd32f450z_eval/support/openocd.cfg [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2022, Rtone.
```
Comment (nandojve) [inline boards/arm/gd32f450z_eval/support/openocd.cfg:1]: Don`t forget to keep rights if you copy and adapted this part.

```diff
+# Copyright (c) 2021, ATL-Electronics
+# Copyright (c) 2021, Teslabs Engineering S.L.
# Copyright (c) 2022, Rtone.
```

**Review by nandojve — CHANGES_REQUESTED** [review]

File: boards/arm/gd32f450z_eval/support/openocd.cfg (hunk 4 above)
Comment (sbourdelin) [reply boards/arm/gd32f450z_eval/support/openocd.cfg:3]: oupsy, got it, thx!

**Review by gmarull — DISMISSED** [review]

**Review by nandojve — DISMISSED** [review]

**Comment by sbourdelin** [comment]
Hi @nandojve ,

No indeed, i m gonna take a look

**Comment by sbourdelin** [comment]
Hi @nandojve ,

So, if i'm not misunderstanding, the problem is that the `gigadevice HAL` already use the `SPI_SLAVE` define name for the register configuration [[1](https://github.com/zephyrproject-rtos/hal_gigadevice/blob/main/gd32f4xx/standard_peripheral/include/gd32f4xx_spi.h#L162)] where the `spi_looback` test driver use the same define name to define which slave should be addressed [[2](https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/spi/spi_loopback/src/spi.c#L20)].

Not sure how you want to address this case?
As any other Gigadevice boards using this HAL will suffer from the same problem, i was thinking we could rename the define  `SPI_SLAVE` in the  `spi_looback` to `SPI_SLAVE_NUMBER` which might actually looks more accurate too.
What do you think?

**Comment by gmarull** [comment]
> Hi @nandojve ,
> 
> So, if i'm not misunderstanding, the problem is that the `gigadevice HAL` already use the `SPI_SLAVE` define name for the register configuration [[1](https://github.com/zephyrproject-rtos/hal_gigadevice/blob/main/gd32f4xx/standard_peripheral/include/gd32f4xx_spi.h?rgh-link-date=2022-05-11T14%3A48%3A48Z#L162)] where the `spi_looback` test driver use the same define name to define which slave should be addressed [[2](https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/spi/spi_loopback/src/spi.c?rgh-link-date=2022-05-11T14%3A48%3A48Z#L20)].
> 
> Not sure how you want to address this case? As any other Gigadevice boards using this HAL will suffer from the same problem, i was thinking we could rename the define `SPI_SLAVE` in the `spi_looback` to `SPI_SLAVE_NUMBER` which might actually looks more accurate too. What do you think?

makes sense for a quick solution, also agree with @nandojve suggestion.

**Comment by sbourdelin** [comment]
> However, it will be important open an enhancement to remove all the recursive dependencies related to `hal_gigadevice/<CONFIG_SOC_SERIES>/standard_peripheral)/include/<CONFIG_SOC_SERIES>_libopt.h`. Then link that issue with this PR to allow track the origin.

I'm sorry you lost me there, i'm not sure if you want me to do something about it and i'm even less sure to see the link between the redefine and the recursive dependencies 😅

**Comment by nandojve** [comment]
Hi @sbourdelin ,

No problem, let's try clarify.

> I'm sorry you lost me there, i'm not sure if you want me to do something about it and i'm even less sure to see the link between the redefine and the recursive dependencies sweat_smile

You need to do 2 things:
- Add your proposal to change SPI_SLAVE to SPI_SLAVE_NUMBER_NUMBER to fix CI and get this merged.
- Open an Issue to remind maintainers that there is an issue related to hal_gigadevice/<CONFIG_SOC_SERIES>/standard_peripheral)/include/<CONFIG_SOC_SERIES>_libopt.h.

**Comment by sbourdelin** [comment]
hi @nandojve 

Thanks for the explanations, i added a commit to this PR for the spi test, maybe you want it not link to this PR?
For your suggestion on the Gigadevice HAL, it doesn't seems i can open an issue on the https://github.com/zephyrproject-rtos/hal_gigadevice repository.

Let me know, thanks!

File: tests/drivers/spi/spi_loopback/src/spi.c [hunk 5]
```diff
@@ -17,7 +17,7 @@ LOG_MODULE_REGISTER(main);
 #include <drivers/spi.h>
 
 #define SPI_DRV_NAME	CONFIG_SPI_LOOPBACK_DRV_NAME
-#define SPI_SLAVE	CONFIG_SPI_LOOPBACK_SLAVE_NUMBER
+#define SPI_SLAVE_NUMBER	CONFIG_SPI_LOOPBACK_SLAVE_NUMBER
 #define SLOW_FREQ	CONFIG_SPI_LOOPBACK_SLOW_FREQ
 #define FAST_FREQ	CONFIG_SPI_LOOPBACK_FAST_FREQ
```
Comment (nandojve) [inline tests/drivers/spi/spi_loopback/src/spi.c:22]: Please keep vertikal alignment for CONFIG_*

**Review by nandojve — CHANGES_REQUESTED** [review]

**Comment by nandojve** [comment]
Hi @sbourdelin ,

> For your suggestion on the Gigadevice HAL, it doesn't seems i can open an issue on the https://github.com/zephyrproject-rtos/hal_gigadevice repository.

Please open issue here on ZephyrProject and you can assign to me.

File: tests/drivers/spi/spi_loopback/src/spi.c (hunk 5 above)
Comment (sbourdelin) [reply tests/drivers/spi/spi_loopback/src/spi.c]: sure! done

**Comment by sbourdelin** [comment]
Hi @nandojve ,

> Hi @sbourdelin ,
> 
> > For your suggestion on the Gigadevice HAL, it doesn't seems i can open an issue on the https://github.com/zephyrproject-rtos/hal_gigadevice repository.
> 
> Please open issue here on ZephyrProject and you can assign to me.

Issue is here: https://github.com/zephyrproject-rtos/zephyr/issues/45720
It looks i can't assign it to you directly.

File: boards/arm/gd32f450z_eval/Kconfig [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022, Rtone.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_INIT_PRIORITY
+	int "Board initialization priority"
+	default 50
+	help
+	  Board initialization priority.
```
Comment (gmarull) [inline boards/arm/gd32f450z_eval/Kconfig:8]: where is this used btw?

Comment (sbourdelin) [reply boards/arm/gd32f450z_eval/Kconfig:8]: This is indeed not required, lazy copy paste from the GD32F450I board, my bad.
I'm deleting it thx!

**Review by gmarull — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Pull request merged**